OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -30,6 +33,7 @@ public Attack(Actor self, Target target, WDist minRange, WDist maxRange, bool al
 			Target = target;
 			this.minRange = minRange;
 			this.maxRange = maxRange;
+			rangeModifiers = self.TraitsImplementing<IRangeModifier>();
```

Comment:
Unfortunately, that would prevent range modifier changes from affecting this activity such as if the modifier is conditional on being near an actor to provide a range boost. I am thinking about creating a way for traits to notify activities and other traits of changing such that reevaluation may be required.

I suppose that could be done if this adversely affects performance; however, a note of the resulting bug (activity range not updating with modifiers) would have to be added.